File: GFG_BeginnersDSASheet/Missing in Array/sol.py
class Solution:
    def missingNum(self, arr):
        # code here
        temp=0
        arrSum=0
        leng=len(arr)+2
        
        for x in range(1, leng, 1):
            temp ^= x
            
        for x in arr:
            arrSum ^= x
            
        return temp^arrSum 

File: GFG_BeginnersDSASheet/Missing in Array/sol.java
class Solution {
    int missingNum(int arr[]) {
        // code here
        int all=0;
        int arrSum=0;
        int leng=arr.length;
        
        for(int i=1;i<leng+2;i++){
            all ^= i;
        }
        
        for(int i=0;i<arr.length;i++){
            arrSum ^= arr[i];
        }
        
        return all^arrSum;
    }
}